sdk/servicebus/Microsoft.Azure.ServiceBus/src/Amqp/AmqpConnectionHelper.cs

```diff
@@ -128,7 +128,8 @@ public static AmqpConnectionSettings CreateAmqpConnectionSettings(uint maxFrameS
             {
                 MaxFrameSize = maxFrameSize,
                 ContainerId = containerId,
-                HostName = hostName
+                HostName = hostName,
+                IdleTimeOut = (uint)Constants.DefaultOperationTimeout.TotalMilliseconds
```

Comment:
What was the behavior without the explicit setting?  What's the driver behind this change?   I'm curious on two fronts:

- It likely changes the behavior of the library in a subtle way that current consumers may not expect and could possibly need to be a breaking change.

- I'm curious if this is something that should be applied to Event Hubs as well, or if this is a case that is specific to Service Bus.

Reply:
So essentially, during a network communication issue, when packets are not sent being sent from client to service (or vice versa) this parameter dictates how long should the connection be alive. On windows, the connection gets dropped automatically after sometime. However, in Linux, the connection stays in the `ESTABLISHED` state (using `netstat`) even after several minutes. 
From the customer view, it looks like `MessageSender.Send()` is called, but the operation just times out and throws `ServiceBusTimeoutException` instead of communication exception, since the connection never gets into faulted state.
With this change, we are forcing that 1 minute of idle time on the connection would force it to drop the connection and reestablish when required.

> It likely changes the behavior of the library in a subtle way that current consumers may not expect and could possibly need to be a breaking change.

I wouldn't exactly consider it a breaking change and would think of it as a bugfix. It already works as expected on windows machine, and this would start working on linux as well.

Reply:
> I'm curious if this is something that should be applied to Event Hubs as well, or if this is a case that is specific to Service Bus.

Yes. It should be applied to Event Hubs as well.

Reply:
Thanks, Neeraj.   I appreciate the context.